Automattic/simplenote-android
# Add TagsViewModel to restructure UI and orchestration logic in TagsActivity

Fixes #1371

### Fix
This PR restructure the code of `TagsActivity` to use ViewModels (`TagsViewModel`)  in order to decouple data management logic from the view. 

### Test

#### Automatic Tests
1. Run `TagsActivityTest` inside `androidTest`. It requires to run an on emulator or physical device. 
2. Run `TagsViewModelTest` inside `test`

#### Manual Tests
1. Open Simplenote app
2. Tab on left menu
3. Tab on EDIT tags
4. Try deleting tags without and with notes. It should update the list
5. Try tab on tags to open a dialog to update the tag
6. Try tab on + button to add new tags
7. Try searching tags

### Review
This PR will be reviewed by @ParaskP7 .

### Release
These changes do not require release notes.
